projection/core/src/main/scala/com/lightbend/lagom/internal/projection/ProjectionRegistryActor.scala

```diff
@@ -210,6 +258,8 @@ class ProjectionRegistryActor extends Actor with ActorLogging {
   }
 
   private def updateNameIndex(workerName: WorkerKey, metadata: WorkerCoordinates): Unit = {
+    // eagerly update the local copy before propagating. This prevents race conditions with Changed(XxxUpdateRequest)
+    nameIndexLocalCopy += workerName -> metadata
```

Comment:
eagerly updates `nameIndexLocalCopy`. It's a monotonic map so it's safe to eagerly do this update.